--- terraria/Decompiled/Terraria/Social/Steam/CoreSocialModule.cs
+++ terraria/Terraria/Social/Steam/CoreSocialModule.cs
@@ -1,6 +1,5 @@
 using System;
 using System.Threading;
-using System.Windows.Forms;
 using ReLogic.OS;
 using Steamworks;
 using Terraria.Localization;
@@ -39,7 +38,7 @@
 		}
 		if (!SteamAPI.Init())
 		{
-			MessageBox.Show(Language.GetTextValue("Error.LaunchFromSteam"), Language.GetTextValue("Error.Error"));
+			Console.WriteLine(Language.GetTextValue("Error.LaunchFromSteam"), Language.GetTextValue("Error.Error"));
 			Environment.Exit(1);
 		}
 		IsSteamValid = true;
