--- terraria/Decompiled/Terraria/Localization/LanguageManager.cs
+++ terraria/Terraria/Localization/LanguageManager.cs
@@ -95,7 +95,8 @@
 	private string[] GetLanguageFilesForCulture(GameCulture culture)
 	{
 		Assembly.GetExecutingAssembly();
-		return Array.FindAll(typeof(Program).Assembly.GetManifestResourceNames(), (string element) => element.StartsWith("Terraria.Localization.Content." + culture.CultureInfo.Name) && element.EndsWith(".json"));
+        string[] names = ["", ".Game", ".Items", ".Legacy", ".NPCs", ".Projectiles", ".Town"];
+        return names.Select(x => $"Terraria.Localization.Content.{culture.Name}{x}.json").ToArray();
 	}
 
 	public void SetLanguage(GameCulture culture)
