--- terraria/Decompiled/ReLogic/Localization/IME/FnaIme.cs
+++ terraria/ReLogic/Localization/IME/FnaIme.cs
@@ -1,4 +1,5 @@
 using Microsoft.Xna.Framework.Input;
+using System;
 
 namespace ReLogic.Localization.IME;
 
@@ -14,9 +15,10 @@
 
 	public override uint SelectedCandidate => 0u;
 
-	public FnaIme()
+	public FnaIme(IntPtr windowHandle)
 	{
 		TextInputEXT.TextInput += OnCharCallback;
+		TextInputEXT.WindowHandle = windowHandle;
 	}
 
 	private void OnCharCallback(char key)
@@ -48,6 +50,16 @@
 		}
 	}
 
+	protected override void OnEnable() {
+		base.OnEnable();
+		TextInputEXT.StartTextInput();
+	}
+	protected override void OnDisable()
+	{
+		base.OnDisable();
+		TextInputEXT.StopTextInput();
+	}
+
 	~FnaIme()
 	{
 		Dispose(disposing: false);
