diff --git a/FNA.csproj b/FNA.csproj
index 84be250..86215eb 100644
--- a/FNA.csproj
+++ b/FNA.csproj
@@ -409,7 +409,6 @@
     <Compile Include="src\Vector2.cs" />
     <Compile Include="src\Vector3.cs" />
     <Compile Include="src\Vector4.cs" />
-    <Compile Include="lib\SDL2-CS\src\SDL2.cs" />
     <Compile Include="lib\SDL3-CS\SDL3\SDL3.Legacy.cs" />
     <Compile Include="lib\FAudio\csharp\FAudio.cs" />
     <Compile Include="lib\Theorafile\csharp\Theorafile.cs" />
diff --git a/src/Content/ContentTypeReaderManager.cs b/src/Content/ContentTypeReaderManager.cs
index 5fcf9e1..12f44fc 100644
--- a/src/Content/ContentTypeReaderManager.cs
+++ b/src/Content/ContentTypeReaderManager.cs
@@ -21,6 +21,9 @@ using System.Text.RegularExpressions;
 
 namespace Microsoft.Xna.Framework.Content
 {
+	public class ContentTypeReaderMetaTypeManager {
+		public static Type BackupType;
+	}
 	public sealed class ContentTypeReaderManager
 	{
 		#region Private Variables
@@ -196,6 +199,15 @@ namespace Microsoft.Xna.Framework.Content
 						readerTypeString = PrepareType(readerTypeString);
 
 						Type l_readerType = Type.GetType(readerTypeString);
+						if (l_readerType == null)
+						{
+							if (readerTypeString == "Microsoft.Xna.Framework.Content.ListReader`1[[System.Char, mscorlib]]")
+							{
+								l_readerType = typeof(Microsoft.Xna.Framework.Content.ListReader<System.Char>);
+							} else {
+								l_readerType = ContentTypeReaderMetaTypeManager.BackupType;
+							}
+						}
 						if (l_readerType != null)
 						{
 							ContentTypeReader typeReader;
diff --git a/src/FNAPlatform/FNAPlatform.cs b/src/FNAPlatform/FNAPlatform.cs
index c17454b..08d40eb 100644
--- a/src/FNAPlatform/FNAPlatform.cs
+++ b/src/FNAPlatform/FNAPlatform.cs
@@ -36,7 +36,7 @@ namespace Microsoft.Xna.Framework
 			 * -flibit
 			 */
 
-			bool useSDL3 = Environment.GetEnvironmentVariable("FNA_PLATFORM_BACKEND") == "SDL3";
+			bool useSDL3 = true;
 
 			if (useSDL3)
 			{
diff --git a/src/Game.cs b/src/Game.cs
index 14021ec..ea1f5d2 100644
--- a/src/Game.cs
+++ b/src/Game.cs
@@ -181,7 +181,7 @@ namespace Microsoft.Xna.Framework
 
 		#region Internal Variables
 
-		internal bool RunApplication;
+		public bool RunApplication;
 
 		#endregion
 
