--- terraria/Decompiled/Terraria/Utilities/FileBrowser/NativeFileDialog.cs
+++ terraria/Terraria/Utilities/FileBrowser/NativeFileDialog.cs
@@ -6,11 +6,14 @@
 {
 	public string OpenFilePanel(string title, ExtensionFilter[] extensions)
 	{
+		throw new System.Exception("TODO");
+		/*
 		string[] value = extensions.SelectMany((ExtensionFilter x) => x.Extensions).ToArray();
 		if (nativefiledialog.NFD_OpenDialog(string.Join(",", value), null, out var outPath) == nativefiledialog.nfdresult_t.NFD_OKAY)
 		{
 			return outPath;
 		}
+		*/
 		return null;
 	}
 }
